addons/payment_stripe_sca/static/src/js/stripe_sca.js

```diff
@@ -0,0 +1,111 @@
+odoo.define('payment_stripe_sca.stripe_sca', function (require) {
+"use strict";
+
+var ajax = require('web.ajax');
+var core = require('web.core');
+
+var qweb = core.qweb;
+var _t = core._t;
+
+if ($.blockUI) {
+    // our message needs to appear above the modal dialog
+    $.blockUI.defaults.baseZ = 2147483647; //same z-index as StripeCheckout
+    $.blockUI.defaults.css.border = '0';
+    $.blockUI.defaults.css["background-color"] = '';
+    $.blockUI.defaults.overlayCSS["opacity"] = '0.9';
+}
+
+// require('web.dom_ready');
+// if (!$('.o_payment_form').length) {
+//     return Promise.reject("DOM doesn't contain '.o_payment_form'");
+// }
+
+function displayError(message) {
+    var wizard = $(qweb.render('stripe.error', {'msg': message || _t('Payment error')}));
+    wizard.appendTo($('body')).modal({'keyboard': true});
+    if ($.blockUI) {
+        $.unblockUI();
+    }
+    $("#o_payment_form_pay").removeAttr('disabled');
+}
+
+
+function _redirectToStripeCheckout() {
+    // Open Checkout with further options
+    if ($.blockUI) {
+        var msg = _t("Just one more second, we are redirecting you to Stripe...");
+        $.blockUI({
+            'message': '<h2 class="text-white"><img src="/web/static/src/img/spin.png" class="fa-pulse"/>' +
+                    '    <br />' + msg +
+                    '</h2>'
+        });
+    }
+
+    var paymentForm = $('.o_payment_form');
+    if (!paymentForm.find('i').length) {
+        paymentForm.append('<i class="fa fa-spinner fa-spin"/>');
+        paymentForm.attr('disabled', 'disabled');
+    }
+
+    var stripe = Stripe($("input[name='stripe_key']").val());
+
+    stripe.redirectToCheckout({
+        sessionId: $("input[name='session_id']").val()
+    }).then(function (result) {
+        if (result.error) {
+            displayError(result.error.message);
+        }
+    });
+}
+
+$('#pay_stripe').on('click', function(e) {
+    // Open Checkout with further options
+    if(!$(this).find('i').length)
+        $(this).append('<i class="fa fa-spinner fa-spin"/>');
+        $(this).attr('disabled','disabled');
+
+    var $form = $(e.currentTarget).parents('form');
+    var acquirer_id = $(e.currentTarget).closest('div.oe_sale_acquirer_button,div.oe_quote_acquirer_button,div.o_website_payment_new_payment');
+    acquirer_id = acquirer_id.data('id') || acquirer_id.data('acquirer_id');
+    if (! acquirer_id) {
+        return false;
+    }
+
+    var so_token = $("input[name='token']").val();
+    var so_id = $("input[name='return_url']").val().match(/quote\/([0-9]+)/) || undefined;
+    if (so_id) {
+        so_id = parseInt(so_id[1]);
+    }
+
+    e.preventDefault();
+    if ($('.o_website_payment').length !== 0) {
```

Comment:
i assume this selector can be used to reject the module at the start (see my comment at beginning of file)

Reply:
I don't think here we can manage this, we have two cases after clicking on pay_stripe, and we have managed stripe button condition, so we don't need this anymore.